cli/cli.go

```diff
@@ -111,70 +107,29 @@ func transformDeprecatedArgs(args []string) []string {
 }
 
 func SetupCLI(args []string) error {
-	runOptions := &runOptionsType{}
-	// There's a bug in github.com/urfave/cli making all commands use
-	// SubCommandHelpTemplate - which has a nasty way of formating command
-	// descriptions
-	cli.SubcommandHelpTemplate = `NAME:
-   {{.HelpName}} {{if .Usage}}- {{.Usage}}{{end}}
-
-USAGE:
-   {{if .UsageText}}{{.UsageText}}{{else}}{{.HelpName}} command` +
-		`{{if .VisibleFlags}} [command options]{{end}} ` +
-		`{{if .ArgsUsage}}{{.ArgsUsage}}{{end}}{{end}}{{if .Description}}
-
-DESCRIPTION:
-   {{.Description}}{{end}}
-
-COMMANDS:{{range .VisibleCategories}}{{if .Name}}
-
-   {{.Name}}:{{range .VisibleCommands}}
-     {{join .Names ", "}}{{"\t"}}{{.Usage}}{{end}}{{else}}{{range .VisibleCommands}}
-   {{join .Names ", "}}{{"\t"}}{{.Usage}}{{end}}{{end}}{{end}}{{if .VisibleFlags}}
-
-OPTIONS:
-   {{range .VisibleFlags}}{{.}}
-   {{end}}{{end}}
-`
-
 	// Filter commandline arguments for backwards compatibility.
 	// FIXME: Remove argument filtering in Mender v3.0
 	args = transformDeprecatedArgs(args)
 
 	app := &cli.App{
-		Before:      runOptions.handleLogFlags,
 		Description: appDescription,
 		Name:        "mender",
-		Usage:       "manage and start the Mender client.",
-		Version:     ShowVersion(),
 	}
-	app.Commands = []cli.Command{
+	app.Commands = []*cli.Command{
 		{
-			Name:  "bootstrap",
-			Usage: "Perform bootstrap and exit.",
-			Flags: []cli.Flag{
-				cli.BoolFlag{
-					Name:        "forcebootstrap, F",
-					Usage:       "Force bootstrap.",
-					Destination: &runOptions.bootstrapForce},
-			},
-			Action: func(ctx *cli.Context) error {
-				if len(ctx.Args()) > 0 {
-					return errors.Errorf(
-						errMsgAmbiguousArgumentsGivenF,
-						ctx.Args().First())
-				}
-				return runOptions.handleCLIOptions(ctx)
-			},
```

Comment:
Why are the arg checks being removed everywhere, is this automatically handled by not specifically requesting positional arguments?

Reply:
I just moved this check further down to the `commonCLIHandler:375` instead of repeating inline in every single closure.